x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx

```diff
@@ -62,13 +61,20 @@ export function ApmServiceContextProvider({
     end,
   });
 
-  const transactionType = getOrRedirectToTransactionType({
+  const currentTransactionType = getTransactionType({
     transactionType: query.transactionType,
     transactionTypes,
     agentName,
-    history,
   });
 
+  // Replace transactionType in the URL in case it is not one of the types returned by the API
+  if (
+    currentTransactionType &&
+    currentTransactionType !== query.transactionType
+  ) {
+    replace(history, { query: { transactionType: currentTransactionType } });
+  }
```

Comment:
Can you put this back into the function? We also have tests for this. I'm assuming they break if you do this.

Reply:
Yeah, I'm aware that there are tests for it. I'm updating the tests to ensure this update has no regression.

Reply:
ok! Can you make sure you're testing the redirect logic still?

Reply:
@dgieselaar, in case it's become overcropped for this PR, would you agree to put back the history as an optional argument?

Reply:
I'm sorry, but I don't follow this. AFAIR the suggestion was for you to extract the part of the function that returns the transaction type, and import that extracted function. That would allow you to keep the test as-is and not making any other changes to a critical part of our application code. Is there anything I'm not seeing that is stopping you from doing that? To clarify, you would have:

- `getOrRedirectToTransactionType`, which is the function we have today
- `getTransactionType`, that only returns the transaction type based on its input, and is called by `getOrRedirectTransactionType`
- you would import `getTransactionType` in your code

Reply:
... I see your point about the ability to test it when it's encapsulated in a function.

Reply:
Ah I did not see your follow up comment on my suggestion. Apologies for the confusion @fkanout. Still, I think it's critical that we have unit tests for `getOrRedirectToTransactionType`, so I suggest we stick to my initial suggestion (as described above).

Reply:
@dgieselaar, it was a bit confusing, but no worries! I applied the suggested implementation